gptel-transient.el

```diff
@@ -185,22 +185,18 @@ which see."
   (transient-parse-suffixes
    'gptel-system-prompt
    (cl-loop for (type . prompt) in gptel-directives
-       with taken
+       ;; Avoid clashes with the custom directive key
+       with unused-keys = (remove ?h (number-sequence ?a ?z))
        with width = (window-width)
        for name = (symbol-name type)
-       for key =
-       (let ((idx 0) pos)
-         (while (or (not pos) (member pos taken))
-           (setq pos (substring name idx (1+ idx)))
-           (cl-incf idx))
-         (push pos taken)
-         pos)
+       for key = (seq-find (lambda (k) (member k unused-keys)) name (seq-first unused-keys))
```

Comment:
The current method tries to provide mnemonic keybindings, using an alternative only in the event of a clash. IIUC this just assigns keys randomly to the directives?

Mnemonic keybindings are useful here (as opposed to just assigning them to 1 - 20) as they can be used by power users without requiring the transient menu to be shown. `(setq transient-show-popup nil)`

Reply:
This does the same but assigns a random (the first) unused key if all characters of the name have already been assigned to other directives, i.e. `seq-find` iterates through the characters of the name and returns the first that is `(member k unused-keys)`, defaulting to `(seq-first unused-keys)` if no key in `name` is unused.

One could think about if it makes sense to work with small and capital letters here, so that two directives `chat` and `complete` can be assigned `c` and `C` instead of `c` and `o`.

Reply:
Thanks for the clarification.

I'm happy with using small letters consistently here.